core/container.go

```diff
@@ -940,7 +1023,7 @@ func locatePath[T *File | *Directory](
 
 	// NB(vito): iterate in reverse order so we'll find deeper mounts first
 	for i := len(container.Mounts) - 1; i >= 0; i-- {
-		mnt := container.Mounts[i]
+		mnt := &container.Mounts[i]
```

Comment:
note that this change was needed to allow `mount.Result = newDir.Result` to be set in `container.WithSymlink`

Reply:
This change is no longer needed since we're going to add a `replaceMount` call.

I'm not sure if we really want it to return a pointer to the actual data, to allow modifications, especially if there's no current use-case.